lib/floe/workflow/intrinsic_function.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    class IntrinsicFunction
+      class << self
+        def klass(payload)
+          function_module, function =
+            payload.match(/^(?<module>\w+)\.(?<function>\w+)\(.*\)$/)
+                   .named_captures.values_at("module", "function")
+
+          case "#{function_module}.#{function}"
+          when "States.StringToJson"
+            Floe::Workflow::IntrinsicFunctions::States::StringToJson
+          else
+            raise NotImplementedError
+          end
+        end
+
+        def value(payload, context, input = {})
+          new(payload).value(context, input)
+        end
+
+        private alias_method :orig_new, :new
+
+        def new(*args)
+          if self == Floe::Workflow::IntrinsicFunction
+            klass(*args).new(*args)
+          else
+            orig_new(*args)
```

Comment:
I think there's a slightly nicer way to do this "intanstiate from the abstract class" pattern without needing to alias_method, but I need to find how we did it elsewhere (Not going to hold up the PR for it)

Reply:
:+1: I pulled this from LinuxAdmin FWIW

Reply:
haha - that's where I was gonna check first :D - let me check in a couple other places, but this is probably fine.